src/main/java/executor/service/service/impl/webDriver/WebDriverInitializerImpl.java

```diff
@@ -0,0 +1,70 @@
+package executor.service.service.impl.webDriver;
+
+import com.google.common.base.Strings;
+import executor.service.config.properties.PropertiesConfig;
+import executor.service.config.properties.PropertiesConstants;
+import executor.service.model.ProxyConfigHolder;
+import executor.service.model.WebDriverConfig;
+import executor.service.service.WebDriverInitializer;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+
+import java.io.File;
+import java.time.Duration;
+import java.util.Properties;
+
+/**
+ * The {@code WebDriverInitializerImpl} class is a factory that produces WebDriver instances.
+ *
+ * @author Oleksii Bondarenko
+ * @version 0.3
+ */
+public class WebDriverInitializerImpl implements WebDriverInitializer {
+    private final Properties properties;
+
+    public WebDriverInitializerImpl(PropertiesConfig propertiesConfig) {
+        this.properties = propertiesConfig.getProperties(PropertiesConstants.WEB_DRIVER);
+    }
+
+    @Override
+    public WebDriver getInstance(WebDriverConfig webDriverConfig, ProxyConfigHolder proxyConfigHolder) {
+        String host = proxyConfigHolder.getProxyNetworkConfig().getHostname();
+        Integer port = proxyConfigHolder.getProxyNetworkConfig().getPort();
+        String username = proxyConfigHolder.getProxyCredentials().getUsername();
+        String password = proxyConfigHolder.getProxyCredentials().getPassword();
```

Comment:
There are no sense to initialize this variables, you can call them from place where they are need

Reply:
these variables are used twice, so this is done for better readability

Reply:
This is first place: proxyPlugin = ChromeProxyPlugin.generate(host, port, username, password);

Where second?

Reply:
Initializing the username and password variables directly in the if is possible in this case, and it depends on your programming style and your code requirements. It is important to keep the readability and logic of the code. But if this is a fundamental issue for you, then write and the team will fix it.

Reply:
I haven't seen any requirements for **username and password** validation anywhere.

Reply:
> I haven't seen any requirements for **username and password** validation anywhere.

I don't say you about **requirements**, I say you how will look condition if next time you have to add validation? In four lines or five, or maybe ten? Also it isn't good for read such long String.emptyOrNull() first you read all function, than read argument. You mean that give more explicit process name is bad? Or  isValidLoginPass does not fully describe what is happening?

Reply:
![image](https://github.com/GFLCourses5/worker/assets/140727960/a573379d-805c-4e12-8f49-6fda4d701e53)
This code is quite readable.

Reply:
> ![image](https://user-images.githubusercontent.com/140727960/274234072-a573379d-805c-4e12-8f49-6fda4d701e53.png) This code is quite readable.

but I said about this:
 if (!Strings.isNullOrEmpty(username) && !Strings.isNullOrEmpty(password))